File: 830. Positions of Large Groups/830.go
package  main

import "fmt"

func main(){
	fmt.Println(largeGroupPositions("abcdddeeeeaabbbcd"))
}

func largeGroupPositions(S string) [][]int {
	var mylist [][]int
	pre := 0
	slen := 0
	for i:=0; i<len(S); i++{
		if S[pre] == S[i]{
			slen ++
		}else{
			if slen>2{
				mylist = append(mylist, []int{pre, i-1})
			}
			slen = 1
			pre = i
		}
	}
	if slen>2{
		mylist = append(mylist, []int{pre, len(S)-1})
	}
	return mylist
}

File: 830. Positions of Large Groups/830.py
class Solution:
    def largeGroupPositions(self, S: str):
        pre = 0
        length = 0
        mylist = list()
        for i in range(len(S)):
            if S[i] == S[pre]:
                length += 1
            else:
                if length > 2:
                    mylist.append([pre,i-1])
                pre = i
                length = 1
        if length > 2:
                    mylist.append([pre,len(S)-1])
        return mylist

    def largeGroupPositions1(self, S):
        i, j, N = 0, 0, len(S)
        res = []
        while i < N:
            while j < N and S[j] == S[i]: j += 1
            if j - i >= 3: res.append((i, j - 1))
            i = j
        return res

a = Solution()
print(a.largeGroupPositions("aaa"))